nway/utils.py

```diff
@@ -91,6 +92,53 @@ def row_col_from_roi(roi):
     return masked
 
 
+def layered_mask_from_rois(rois: List[Any],
+                           shape: Tuple[int, int]) -> np.ndarray:
+    """makes a (nz, *shape) dimensioned array where each layer ("z")
+    has no overlapping ROIs
+
+    Parameters
+    ----------
+    rois: list
+        list of ROIs. The list is the same as "cell_rois" in
+        ExperimentSchema
+    shape: tuple
+        (nrows, ncols) of the destination image
+
+    Returns
+    -------
+    masks: numpy.ndarray
+       dimensions (nz, nrows, ncols)
+
+    """
+    def new_layer():
+        return np.zeros(shape, dtype=np.uint32)
+
+    masks = [new_layer()]
+    for roi in rois:
+        tmp = new_layer()
+        tmp[roi['y']:(roi['y'] + roi['height']),
+            roi['x']:(roi['x'] + roi['width'])] = np.uint32(roi['mask_matrix'])
+        tmp *= roi['id']
+        bool_tmp = tmp != 0
+        placed = False
+        for imask in range(len(masks)):
```

Comment:
Might be more readable to have:

```
for layer_indx, layer in enumerate(mask_layers):
```

Not changing is fine too.

Reply:
I had this previously, but in this case `layer` is not mutable. `masks[index]` is mutable. So, I went with this way.

Reply:
Ah, I see. Good call.